patches/net/minecraft/server/network/ServerGamePacketListenerImpl.java.patch

```diff
@@ -17,14 +17,37 @@
              }
  
              entity.absMoveTo(d3, d4, d5, f, f1);
-+            this.player.absMoveTo(d3, d4, d5, this.player.getYRot(), this.player.getXRot()); // Forge - Resync player position on vehicle moving
++            resyncPlayerWithVehicle(entity); // Neo - Resync player position on vehicle moving
              boolean flag3 = serverlevel.noCollision(entity, entity.getBoundingBox().deflate(0.0625));
              if (flag && (flag2 || !flag3)) {
                 entity.absMoveTo(d0, d1, d2, f, f1);
-+               this.player.absMoveTo(d3, d4, d5, this.player.getYRot(), this.player.getXRot()); // Forge - Resync player position on vehicle moving
++               resyncPlayerWithVehicle(entity); // Neo - Resync player position on vehicle moving
                 this.send(new ClientboundMoveVehiclePacket(entity));
                 return;
              }
+@@ -443,6 +_,22 @@
+       }
+    }
+ 
++   private void resyncPlayerWithVehicle(Entity vehicle) {
++      Vec3 oldPos = this.player.position();
++      float yRot = this.player.getYRot();
++      float xRot = this.player.getXRot();
++      float yHeadRot = this.player.getYHeadRot();
++      vehicle.positionRider(this.player);
```

Comment:
For better readability, put blank lines before and after this call to `positionRider`, to separate it visibly from the code which stores the old player position & rotation values and from the code managing the old rotation and position.